pkg/kbagent/util/event.go

```diff
@@ -70,29 +73,59 @@ func createEvent(reason string, message string) *corev1.Event {
 			Component: proto.ProbeEventSourceComponent,
 			Host:      nodeName(),
 		},
-		FirstTimestamp:      metav1.Now(),
-		LastTimestamp:       metav1.Now(),
+		FirstTimestamp:      now,
+		LastTimestamp:       now,
 		EventTime:           metav1.NowMicro(),
 		ReportingController: proto.ProbeEventReportingController,
 		ReportingInstance:   podName(),
 		Action:              reason,
 		Type:                "Normal",
+		Count:               1,
 	}
 }
 
-func sendEvent(event *corev1.Event) error {
+func createOrUpdateEvent(reason, message string) error {
 	clientSet, err := getK8sClientSet()
 	if err != nil {
 		return err
 	}
+	eventsClient := clientSet.CoreV1().Events(namespace())
+
+	eventName := generateEventName(reason, message)
+	event, err := eventsClient.Get(context.Background(), eventName, metav1.GetOptions{})
```

Comment:
I think the Get() call should be part of the retry logic. For example, the call may fail due to temporary network failures, retry can help in this case.